File: packages/ai-config/src/io/configValidator.ts
import { RuntimeConfigSchema } from '../schema/runtime.schema.js';

export function validateRuntimeConfig(config: unknown) {
  return RuntimeConfigSchema.parse(config);
}

export function safeValidateRuntimeConfig(config: unknown) {
  const result = RuntimeConfigSchema.safeParse(config);
  if (result.success) {
    return { ok: true as const, data: result.data };
  }
  return {
    ok: false as const,
    errors: result.error.issues.map((issue) => ({
      path: issue.path.map((entry) => String(entry)).join('.'),
      message: issue.message,
    })),
  };
}

File: packages/ai_config/io/config_validator.py
from ..schema.runtime_schema import RuntimeConfigSchema


def validate_runtime_config(config: dict) -> RuntimeConfigSchema:
    return RuntimeConfigSchema(**config)


def safe_validate_runtime_config(config: dict) -> dict:
    try:
        data = RuntimeConfigSchema(**config)
        return {"ok": True, "data": data}
    except Exception as exc:
        return {"ok": False, "error": str(exc)}
